modules/comparison-of-solutions-to-provision-node-local-storage.adoc

```diff
@@ -0,0 +1,172 @@
+
+.Comparison of the support for storage types and filesystems
+[cols="6a,5a,5a,5a",options="header"]
+|====
+
+|Functionality|LVM Storage|LSO |HPP
+
+|Support for block storage |Yes |Yes |No
+
+|Support for file storage |Yes |Yes |Yes
+
+|Support for object storage ^[1]^ |No |No |No
+
+|Available filesystems |`ext4`, `xfs` |`ext4`, `xfs` |Any mounted system available on the node is supported.
+
+|====
+[.small]
+--
+1. None of the solutions ({lvms}, LSO, and HPP) provide support for object storage. Therefore, an S3 object storage solution, such as `MultiClusterGateway` from the Red Hat OpenShift Data Foundation, is needed. All of the solutions can serve as underlying storage providers for the S3 object storage solutions.
+--
+
+[id="comparing-core-functionalities"]
+== Comparison of the support for core functionalities
+
+The following table compares how {lvms}, Local Storage Operator (LSO), and HostPath Provisioner (HPP) support core functionalities for provisioning local storage:
+
+.Comparison of the support for core functionalities
+[cols="6a,5a,5a,5a",options="header"]
+|====
+
+|Functionality|LVM Storage|LSO |HPP
+
+|Support for automatic file system formatting |Yes |Yes |N/A
+
+|Support for dynamic provisioning |Yes |No |No
+
+|Support for using software Redundant Array of Independent Disks (RAID) arrays 
+|Yes 
+
+Supported on 4.15 and higher versions.
+
+|Yes |Yes
+
+|Support for transparent disk encryption |Yes 
+
+Supported with 4.16 and higher versions.
+
+|Yes |Yes
+
+|Support for volume based disk encryption |No |No |No
+
+|Support for disconnected installation |Yes |Yes |Yes
+
+|Support for PVC expansion |Yes |No |No
+
+|Support for volume snapshots and volume clones |Yes |No |No
+
+|Support for thin provisioning |Yes
+
+Devices are thin-provisioned by default.
+
+|Yes
+
+You can configure the devices to point to thin-provisioned volumes
+
+|Yes
+
+You can configure a path to point to thin-provisioned volumes.
+
+|Support for automatic disk discovery and setup |Yes
+
+Automatic disk discovery is available during installation and runtime. You can also dynamically add the disks to the storage classes to increase the storage capacity.
+
+|Technology Preview
+
+Automatic disk discovery is available during installation. 
+
+|No
+
+|====
+
+[id="comparing-performance-and-isolation-boundary"]
+== Comparison of performance and isolation capabilities
+
+The following table compares the performance and isolation capabilities of {lvms}, Local Storage Operator (LSO), and HostPath Provisioner (HPP) in provisioning local storage.
+
+.Comparison of performance and isolation capabilities
+[cols="5a,6a,6a,6a",options="header"]
+|====
+
+|Functionality|LVM Storage|LSO |HPP
+
+|Performance
+
+|I/O speed is shared for all workloads that use the same storage class. 
+
+Block storage allows direct I/O operations. 
+
+Thin-provisioning can affect the performance. 
+
+|I/O speed is shared for all workloads that use the same storage class. 
```

Comment:
(I might be wrong, but) If the provisioning is based on disks(not partition), I think the I/O is not shared？

Reply:
@jakobmoellerdev @brandisher 
Could you PTAL?

Reply:
For thin provisioning based on volume group, I/O is shared either way. This is because one storage class maps to one volume group. This volume group will get I/O from all pods that a have a volume mounted on it. 

However, this does not directly translate to shared I/O from the pods: a linear volume group might have one pod use data blocks from disk 1 and one pod from disk 2 so they might not share I/O in that layout. 

TLDR: under certain circumstances I/O is not shared, but we do not want to expose this abstraction to the users so it is safer to say that I/O is shared for all workloads (even if it really isnt under some border cases)

Reply:
@duanwei33 
For LSO, how about changing it to "I/O speed is not shared for workloads."
Should we add more info to provide better context?

Hi @gcharot ,
Could you PTAL?
Thanks!

Reply:
Just to understand correctly, I was expecting LSO to use LocalVolume to setup a StorageClass. Within this storage class you use one to many devices. However in this storageclass, I/O between workloads is shared. If there is only one disk then to avoid sharing for workloads, there must be one sc per workload, right?

Reply:
Hi @jakobmoellerdev, local storage could only provide static provisioning, we could define disks or partitions in LSO CR, and it transfers each disk or partition to one pv (the pv number based on how many disks/partitions are detected), and here the storageclass is also created but it looks like a label to get bound with a pv when using the PVC inside the workload.  
So one common use case is:
- user creates PVC1 with storageclass (created by LSO CR), the PVC1 get bound with PV1(disk1) and it is consumed by workload1
- user creates PVC2 with the same storageclass,  the PVC2 get bound with PV2(disk2) and it is consumed by workload2

Here workload1 and workload2 are using two different disks, so I guess the I/O is not shared.

> I/O speed is shared for all workloads that use the same storage class. 


cc @jsafrane @gnufied May I get some help/clarification if I wrote something wrong :(

Reply:
Updated it to "I/O depends on LSO configuration" as suggested by the QE @duanwei33